File: back/src/main/java/com/back/domain/chat/chatRoomMessage/event/ChatMessageSentEvent.java
package com.back.domain.chat.chatRoomMessage.event;

import com.back.domain.chat.chatRoomMessage.dto.RedisChatMessageDto;

import java.util.List;
import java.util.UUID;

public class ChatMessageSentEvent {
    
    private final RedisChatMessageDto messageDto;
    private final List<BroadcastTarget> targets;

    public ChatMessageSentEvent(RedisChatMessageDto messageDto, List<BroadcastTarget> targets) {
        this.messageDto = messageDto;
        this.targets = targets;
    }

    public RedisChatMessageDto getMessageDto() {
        return messageDto;
    }
    public List<BroadcastTarget> getTargets() { return targets; }

    public record BroadcastTarget(UUID participantId, String memberId, boolean isBot) {}

}

File: back/src/main/kotlin/com/back/domain/chat/chatRoomMessage/event/ChatMessageSentEvent.kt
package com.back.domain.chat.chatRoomMessage.event

import com.back.domain.chat.chatRoomMessage.dto.RedisChatMessageDto
import java.util.UUID

class ChatMessageSentEvent(
    val messageDto: RedisChatMessageDto,
    val targets: List<BroadcastTarget>
) {
    data class BroadcastTarget(val participantId: UUID, val memberId: String, val isBot: Boolean)
}
